crates/bmp-pkt/src/lib.rs

```diff
@@ -1028,6 +1028,27 @@ pub struct PeerKey {
     bgp_id: Ipv4Addr,
 }
 
+impl Hash for PeerKey {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.peer_address.hash(state);
+        self.peer_type.get_type().hash(state);
+        self.rd.hash(state);
+        self.asn.hash(state);
+        self.bgp_id.hash(state);
+    }
+}
+impl PartialEq<Self> for PeerKey {
+    fn eq(&self, other: &Self) -> bool {
+        self.peer_address.eq(&other.peer_address)
+            && std::mem::discriminant(&self.peer_type) == std::mem::discriminant(&other.peer_type)
```

Comment:
if i do `self.peer_type == other.peer_type`, when both enums are the same variant it will compare the internal values as well. with `std::mem::discriminant`, we only compare the enum variant regardless of the fields of the variant (which are the flags we want to ignore here)